data.rs

```diff
@@ -0,0 +1,170 @@
+use crate::PSP22Error;
+use ink::{
+    prelude::{vec, vec::Vec},
+    primitives::AccountId,
+    storage::Mapping,
+};
+
+pub enum PSP22Event {
+    Transfer(Option<AccountId>, Option<AccountId>, u128),
+    Approval(AccountId, AccountId, u128),
+}
+
+#[ink::storage_item]
+#[derive(Debug, Default)]
+pub struct PSP22Data {
+    total_supply: u128,
+    balances: Mapping<AccountId, u128>,
+    allowances: Mapping<(AccountId, AccountId), u128>,
+}
```

Comment:
It'd be good to have comments (`///`) on pub elements. Especially this is supposed to be an exemplary usage.

Reply:
I'll add nice comments and extensive readme in a different PR